### PR TITLE
ICU-13668 C Demo Autobuild

### PR DESCRIPTION
**Pull request by srl295**
General update URL: 
https://unicode-org.atlassian.net/browse/ICU-13668

Draft work to move icu4c to be more buildable

### CODE REVIEW

**Comment by sffc** [comment]
There's a lot of stuff in this CL; anything in particular you want me to look at?

**Comment by srl295** [comment]
> There's a lot of stuff in this CL; anything in particular you want me to look at?

at the build process, and a quick check over the changes to see if there's anything that's a problem?

File: icu-kube/docker.d/icu4c-demos/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+USER build
+# VOLUME /home/build
+RUN mkdir /home/build/build /home/build/src
+WORKDIR /home/build/build
+ADD . /home/build/src/
+ENV PREFIX=/home/build/icu
+RUN ~/src/configure  --prefix=${PREFIX} ICU_CONFIG=$(which icu-config) 
+RUN make all OBS= && make install OBS= DESTDIR=${PREFIX}
+#make -k DEPS= clean && make -k DEPS= distclean && sudo apt-get purge -y git subversion python3 doxygen zip curl g++ && sudo apt-get -y autoremove && sudo apt-get clean -y
```
Comment (sffc) [inline icu-kube/docker.d/icu4c-demos/Dockerfile]: Did you mean to delete this?

File: icu-kube/docker.d/icu4c-demos/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# © 2016 and later: Unicode, Inc. and others.
+# License & terms of use: http://www.unicode.org/copyright.html
+
+FROM alpine:latest as build
```
Comment (sffc) [inline icu-kube/docker.d/icu4c-demos/Dockerfile]: Hmm, haven't used Alpine Linux before

File: icu-kube/docker.d/icu4c-demos/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ENV HOME /home/build
+
+RUN apk --update add gcc make python g++ ccache valgrind doxygen tar zip curl wget git bash bsd-compat-headers
+RUN addgroup build && adduser -g "Build user" -h $HOME -S -G build -D -s /bin/sh build
+
+## Change this to a file:///mnt/icu/blah.tgz to replace, or another URL.
+ARG ICU_PATH
+ENV ICU_PATH=${ICU_PATH}
+
+RUN if [ $ICU_PATH = "system" ]; then apk --update add icu-dev; else curl -L $ICU_PATH | (cd / ; tar xvfpz - --strip-components 2 ); fi
```
Comment (sffc) [inline icu-kube/docker.d/icu4c-demos/Dockerfile]: So... this installs libicu in `/lib`, either via the package installer or via extracting the ICU tarball at the root of the filesystem.  Seems hacky but I guess it works and makes things easier further down.

File: icu-kube/docker.d/icu4c-demos/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+# VOLUME /mnt/icu
+RUN icuinfo | tee /tmp/icuinfo.txt
+
+USER build
+# VOLUME /home/build
+RUN mkdir /home/build/build /home/build/src
+WORKDIR /home/build/build
+ADD . /home/build/src/
+ENV PREFIX=/home/build/icu
+RUN ~/src/configure  --prefix=${PREFIX} ICU_CONFIG=$(which icu-config)
```
Comment (sffc) [inline icu-kube/docker.d/icu4c-demos/Dockerfile]: I thought we weren't using icu-config any more

File: icu-kube/docker.d/icu4c-demos/Dockerfile (hunk 1 above)
Comment (srl295) [reply icu-kube/docker.d/icu4c-demos/Dockerfile]: it can reduce the size of that layer

File: icu-kube/docker.d/icu4c-demos/Dockerfile (hunk 2 above)
Comment (srl295) [reply icu-kube/docker.d/icu4c-demos/Dockerfile]: it's nice and tiny

File: icu-kube/docker.d/icu4c-demos/Dockerfile (hunk 3 above)
Comment (srl295) [reply icu-kube/docker.d/icu4c-demos/Dockerfile]: That's what the ICU tarball is for.

File: icu-kube/docker.d/icu4c-demos/Dockerfile (hunk 4 above)
Comment (srl295) [reply icu-kube/docker.d/icu4c-demos/Dockerfile]: we're definitely using it. The configure script in icu-demos still depends on it.  If we can merge this PR i can continue work…

**Pull request merged**